src/common/helpers/sharepoint/sharepoint.test.js

```diff
@@ -111,118 +138,181 @@ const mockFile = {
-    mockUploadFile
-      .mockResolvedValueOnce(undefined)
-      .mockRejectedValueOnce(new Error('upload failed'))
-
-    const response = await sharePointApplicationHandler(
-      mockRequestWithFile,
-      testReferenceNumber
-    )
-
-    expect(uploadFile).toHaveBeenCalled()
-    expect(response).toEqual({
-      error: {
-        errorCode: 'FILE_UPLOAD_FAILED__BIOSECURITY_MAP',
-        statusCode: statusCodes.serverError
-      }
+        expect(sendEmailToCaseWorker).toHaveBeenCalledWith({
+          content: 'Sharepoint notification email content'
+        })
+        expect(result).toBeUndefined()
+      })
+
+      it('should upload application html and biosecurity map to SharePoint and send emails to applicant and case worker', async () => {
+        const mockFileData = { ...mockFile, fileSizeInMB: 5 }
+        mockFetchFile.mockResolvedValue(mockFileData)
+        mockUploadFile.mockResolvedValue(undefined)
+
+        const response = await processApplication(
+          mockQueuedApplicationDataWithFile
+        )
+
+        expect(mockGenerateHtmlBuffer).toHaveBeenCalledWith(
+          mockRequestWithFile.payload,
+          testReferenceNumber
+        )
+        expect(uploadFile).toHaveBeenCalledTimes(2)
+        expect(uploadFile).toHaveBeenNthCalledWith(
+          1,
+          testReferenceNumber,
+          `${testReferenceNumber}_Submitted_Application.html`,
+          Buffer.from('<html>Mock HTML</html>')
+        )
+        expect(uploadFile).toHaveBeenNthCalledWith(
+          2,
+          testReferenceNumber,
+          `${testReferenceNumber}_Biosecurity_Map.pdf`,
+          mockFileData.file
+        )
+
+        expect(createSharepointItem).toHaveBeenCalledWith(
+          mockRequestWithFile.payload,
+          testReferenceNumber
+        )
+
+        expect(sendEmailToCaseWorker).toHaveBeenCalledWith({
+          content: 'Sharepoint notification email content'
+        })
+
+        expect(response).toBeUndefined()
+      })
+
+      it('should throw and log if uploadSubmittedApplication fails', async () => {
+        mockUploadFile.mockRejectedValue(new Error('upload failed'))
+        await expect(
+          processApplication(mockQueuedApplicationData)
+        ).rejects.toThrow('upload failed')
+        expect(mockLoggerWarn).toHaveBeenCalledWith(
+          'Failed to upload submitted application to SharePoint: upload failed'
+        )
+      })
+
+      it('should throw and log if uploadBiosecurityMap fails', async () => {
+        mockUploadFile
+          .mockResolvedValueOnce(undefined)
+          .mockRejectedValueOnce(new Error('biosecurity upload failed'))
+        await expect(
+          processApplication(mockQueuedApplicationDataWithFile)
+        ).rejects.toThrow('biosecurity upload failed')
+        expect(mockLoggerWarn).toHaveBeenCalledWith(
+          'Failed to upload biosecurity map to SharePoint: biosecurity upload failed'
+        )
```

Comment:
Same comment as above: Should we assert that - in this case - we don't get a sharepoint item or notification email here?

Reply:
done